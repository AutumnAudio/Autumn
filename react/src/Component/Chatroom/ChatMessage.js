import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import PersonIcon from '@material-ui/icons/Person';

const styles = {
    personIcon: {
        float: 'left',
        fontSize: '47px',
        color: '#2699FB',
    },
    nameDiv: {
        fontWeight: 'bold',
        marginLeft: '55px',
        paddingTop: '5px',
    },
    textDiv: {
        marginLeft: '55px',
        marginTop: '5px',
        fontWeight: '300',
        wordWrap: 'break-word',
    },
    chatMessageList: {
        margin: '20px',
    },
};

const ChatMessage = (props) => {
    const { classes, message } = props;
    return (
        <div className={classes.chatMessageList}>
            <PersonIcon className={classes.personIcon}/>
            <div className={classes.nameDiv}>{message.username}</div>
            <div className={classes.textDiv}>{message.message}</div>
        </div>
    )
    
}

ChatMessage.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ChatMessage)