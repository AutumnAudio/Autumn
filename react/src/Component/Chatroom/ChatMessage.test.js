import React from 'react'
import renderer from 'react-test-renderer'
import ChatMessage from './ChatMessage'

test('should be able to render chat', () => {
    const chatMock = {username: 'testUser',
                      message: 'some message'}
    const component = renderer.create(
        <ChatMessage message={chatMock} />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
