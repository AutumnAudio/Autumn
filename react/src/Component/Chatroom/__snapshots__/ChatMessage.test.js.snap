// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render chat 1`] = `
<div
  className="ChatMessage-chatMessageList-4"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root ChatMessage-personIcon-1"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
  <div
    className="ChatMessage-nameDiv-2"
  >
    testUser
  </div>
  <div
    className="ChatMessage-textDiv-3"
  >
    some message
  </div>
</div>
`;
